test/unit/math/opencl/kernel_generator/append_test.cpp

```diff
@@ -0,0 +1,192 @@
+#ifdef STAN_OPENCL
+
+#include <stan/math/opencl/kernel_generator.hpp>
+#include <stan/math/opencl/matrix_cl.hpp>
+#include <stan/math/opencl/copy.hpp>
+#include <stan/math/prim/fun.hpp>
+#include <Eigen/Dense>
+#include <gtest/gtest.h>
+
+#define EXPECT_MATRIX_NEAR(A, B, DELTA) \
+  for (int i = 0; i < A.size(); i++)    \
+    EXPECT_NEAR(A(i), B(i), DELTA);
```

Comment:
We really need to move this EXPECT_MATRIX_NEAR to a util file. This is getting ridiculous to add this all the time. Guilty as well. But that is a separate PR.